controllers/barbicanapi_controller.go

```diff
@@ -326,6 +327,45 @@ func (r *BarbicanAPIReconciler) generateServiceConfigs(
 		"EnableSecureRBAC": instance.Spec.EnableSecureRBAC,
 	}
 
+	// Checking if there's an HSM.
+	pkcs11 := instance.Spec.PKCS11
```

Comment:
this will fail when there is no PKCS11 element because this is optional.  You need to check  
if (len(instance.Spec.PKCS11) > 0)  or check for !=nil

As a result , you can use this ^^ and you dont need the HSMEnabled flag.